client/network/src/protocol.rs

```diff
@@ -190,6 +197,8 @@ pub struct Protocol<B: BlockT, H: ExHashT> {
 	propagate_timeout: Pin<Box<dyn Stream<Item = ()> + Send>>,
 	/// Pending list of messages to return from `poll` as a priority.
 	pending_messages: VecDeque<CustomMessageOutcome<B>>,
+	/// Pending extrinsic verification tasks.
+	pending_transactions: VecDeque<(PeerId, TransactionImportFuture)>,
```

Comment:
This should have maximum size

Reply:
Sorry, forgot to push newest branch, there is no VecDeque now

Reply:
It is still unbound

Reply:
It unbounded spawned 2 new tasks before, now it is at least one.

Reply:
I don't see any tasks spawned in this code before. If there was an issue somewhere else, moving it here is still bad enough. Let's just put a bound on it.

Reply:
https://github.com/paritytech/substrate/pull/5939/files#diff-09e11dbb6a6e80c9aa5a56ebc6550873L694

and any call to `submit_one` is also spawns new task in other thread pool

Reply:
It is bounded by reputation now
Every time we put something in this `pending_transactions`, the peer reputation decreases

Reply:
So what's the maximum size in bytes then?

Reply:
This depends on peer reputation
If he is a good fellow, he can send spam us a bit more before get banned.

I don't know what negative reputation results in ban, but whatever this number is, divided by 16

Reply:
Relying on side effects of the reputation system is too flaky here. The reputation changes need to be per byte btw. Why not put an explicit cap?

Reply:
Added hardcoded limit, thanks!